lib/matplotlib/backend_bases.py

```diff
@@ -2149,6 +2151,38 @@ def print_figure(
                 self.figure.set_canvas(self)
             return result
 
+    def _check_savefig_extra_args(self, kwargs):
```

Comment:
Can you explain *briefly* what this does?  at least at first sight it isn't obvious, I guess I could work it out but heh :)

Reply:
This is basically a copy of `warn_external`, but instead of determining stack level, it determines the name of the last Matplotlib-based function to be called, so it can provide the warning for the user-level function name.

Reply:
Added a docstring.